### PR TITLE
`v11.x.x` Updates to `Tag`

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/24

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by yurm04** [comment]
@kyledurand  caught up with @sarahill and confirmed that the tokens used in the new Tag will have updated values. The styling should all be correct once those changes go through but please double check before merging

File: polaris-react/src/components/Tag/Tag.stories.tsx [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
+  return (
+    <Card>
+      <LegacyStack>
+        <Tag>Wholesale</Tag>
+      </LegacyStack>
+    </Card>
+  );
+}
+
+export function Disbled() {
```
Comment (kyledurand) [inline polaris-react/src/components/Tag/Tag.stories.tsx]:
```suggestion
export function Disabled() {
```

**Comment by sarahill** [comment]
Noticed some jumpiness on the clickable tag hover state 
tag-hover-jump

**Comment by kyledurand** [comment]
Closing for https://github.com/Shopify/polaris/pull/9324

**Pull request closed without merging**